Tools/TestWebKitAPI/Tests/WebKitCocoa/WKHTTPCookieStore.mm

```diff
@@ -817,6 +817,30 @@ static bool areCookiesEqual(NSHTTPCookie *first, NSHTTPCookie *second)
     TestWebKitAPI::Util::run(&done);
 }
 
+TEST(WKHTTPCookieStore, SetSameSiteNoneCookie)
+{
+    RetainPtr<WKHTTPCookieStore> httpCookieStore = [WKWebsiteDataStore defaultDataStore].httpCookieStore;
+    RetainPtr cookie = [NSHTTPCookie cookieWithProperties:@{
+        NSHTTPCookiePath: @"/",
+        NSHTTPCookieName: @"a",
+        NSHTTPCookieValue: @"b",
+        NSHTTPCookieDomain: @"127.0.0.1",
+        NSHTTPCookieSameSitePolicy: @"none"
+    }];
+
+    clearCookies([WKWebsiteDataStore defaultDataStore]);
+
+    finished = false;
+    [httpCookieStore.get() setCookie:cookie.get() completionHandler:^{
+        [httpCookieStore.get() getAllCookies:^(NSArray<NSHTTPCookie *> *cookies) {
+            EXPECT_EQ(1u, cookies.count);
+            EXPECT_STREQ("none", cookies.firstObject.sameSitePolicy.UTF8String);
+            finished = true;
+        }];
+    }];
+    TestWebKitAPI::Util::run(&finished);
```

Comment:
Could you call clearCookies again after this test to leave no state behind?
Also, is it worth adding the same test with a non-persistent data store?
Also, could we test the actual functionality of a same-site:none cookie and see that it is sent correctly over the network?

Reply:
> Could you call clearCookies again after this test to leave no state behind?
Yes.
> Also, is it worth adding the same test with a non-persistent data store?
Sure.
> Also, could we test the actual functionality of a same-site:none cookie and see that it is sent correctly over the network?
Can you clarify what you mean by this? Do you want me to do that in this test? There are several existing layout tests that verify SameSite=None cookies are sent correctly over the network.